application/controllers/admin/DataEntry.php

```diff
@@ -1456,6 +1529,24 @@ public function update()
                 continue;
             }
             $thisvalue = Yii::app()->request->getPost($fieldname, '');
+            // For questions, if the "Unseen" checkbox is checked, we must set the field to null.
+            // There are some special cases we need to handle.
+            if ($irow['type'] == Question::QT_R_RANKING) {
+                $unseenFieldName = "unseen:" . $irow['sid'] . 'X' . $irow['gid'] . 'X' . $irow['qid'];
+            } elseif ($irow['type'] == Question::QT_P_MULTIPLE_CHOICE_WITH_COMMENTS) {
+                // Remove trailing "comment" from the fieldname, if present
+                $unseenFieldName = "unseen:" . preg_replace('/comment$/', '', $fieldname);
+            } else {
+                $unseenFieldName = "unseen:" . $fieldname;
+            }
+            if (!empty($irow['title']) && Yii::app()->request->getPost($unseenFieldName, false)) {
+                // Throw an error if "unseen" is checked but the field is not empty. This should never happen.
```

Comment:
Send a flash error seems better in my opinion ?

1. If value + unseen is set : save the value
2. But send a alert danger message 

```
if (!empty($irow['title']) && Yii::app()->request->getPost($unseenFieldName, false)) {
    // Throw an error if "unseen" is checked but the field is not empty: javascript didn't work good here
    if ($thisvalue === '') {
        $oResponse->$fieldname = null;
    } else {
        App()->addFlash("Question {$irow['title']} was set to unseen but have value. Valuse was saved", 'danger');
    }
    continue;
}
```

Else a JS speed issue or anything other broken : user loose all data sent …